lib/pure/sugar.nim

```diff
@@ -207,6 +207,8 @@ macro capture*(locals: varargs[typed], body: untyped): untyped {.since: (1, 1).}
   let locals = if locals.len == 1 and locals[0].kind == nnkBracket: locals[0]
                else: locals
   for arg in locals:
+    if arg.strVal == "result":
```

Comment:
But this is not correct:

```nim

import sugar

proc begin() =
  var result = 0
  capture result:
    echo 1+1
  result

```

Reply:
This code doesn't compile too for the same reason.

Reply:
And so does this:
```nim
import sugar


var result = 0
capture result:
  echo 1+1
```

Reply:
For what reason? The point is that "result" is only special for the ``nskResult`` symbol kind.

Reply:
`capture` rewrites the variable name as the parameter name of the functions with `auto` return type. And `result`  can't be used as parameters name with `auto` return type.

```nim
# redefinition of 'result';
proc hello(result: int): auto = 
  discard
```

"result" this name causes redefinition errors.

Reply:
That should be fixed instead IMO

Reply:
I think it is reasonable for function with auto type can't use `result` as parameter name. 

We could change `capture` if people have a better solution.